avocado/plugins/plugins.py

```diff
@@ -58,7 +72,11 @@ def run(self, config):
         for plugins_active, msg in plugin_types:
             LOG_UI.info(msg)
             plugin_matrix = []
-            for plugin in sorted(plugins_active, key=lambda x: x.name):
+            if config.get('plugins.ordered_list'):
+                sorted_plugins = plugins_active.get_extentions_by_priority()
+            else:
+                sorted_plugins = plugins_active.get_extentions_by_name()
```

Comment:
By looking at this, seems to me this kind of logic should be part of `plugins_active` object. Something like this:

`plugins_active.get_ordered()`, so we could have a better reuse. However I'm not sure if that is possible to access the config dict from there. So if not, feel free to ignore this comment.

Reply:
IMO, it might be quite tricky to access the config dict from `ExtensionManager` and also it already has `get_extentions_by_priority` and `get_extentions_by_name` so I do  not see reason to have those in another method `get_ordered`. So I wouldn't change this.